src/workerd/io/actor-cache.c++

```diff
@@ -2086,16 +2083,6 @@ void ActorCache::putImpl(Lock& lock, kj::Own<Entry> newEntry,
     // Inherit gap state.
     newEntry->gapIsKnownEmpty = slot->gapIsKnownEmpty;
 
-    if (slot->isDirty()) {
```

Comment:
The reason we can get rid of this (i.e. not inherit the counted delete) is because the `CountedDelete` actually takes a strong reference to the Entry.

In the case the `CountedDelete` _doesn't_ take a strong reference (i.e. Entry is `PRESENT`), we count the delete early, then the new `ABSENT` entry ends up in the dirtyList and we do a regular delete. If we subsequently overwrite with a `put()` before the entry is flushed, then we don't need to bother doing the delete and we replace the `ABSENT` entry (removing from the dirtyList) with the new `PRESENT` one. Since we already contributed to the count, it's effectively the same as if we had done the delete followed by the put.

Reply:
@a-robinson this explains why https://github.com/cloudflare/workerd/pull/1814#discussion_r1540158815 shouldn't be much of a concern.